File: baseLib/src/main/java/me/ycdev/android/lib/common/tracker/WeakTracker.java
package me.ycdev.android.lib.common.tracker;

import me.ycdev.android.lib.common.utils.WeakListenerManager;

public abstract class WeakTracker<IListener> extends WeakListenerManager<IListener> {
    protected abstract void startTracker();
    protected abstract void stopTracker();

    @Override
    protected void onFirstListenerAdd() {
        startTracker();
    }

    @Override
    protected void onLastListenerRemoved() {
        stopTracker();
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/tracker/WeakTracker.kt
package me.ycdev.android.lib.common.tracker

import me.ycdev.android.lib.common.manager.ListenerManager

abstract class WeakTracker<IListener : Any> : ListenerManager<IListener>(true) {
    protected abstract fun startTracker()
    protected abstract fun stopTracker()

    override fun onFirstListenerAdd() {
        startTracker()
    }

    override fun onLastListenerRemoved() {
        stopTracker()
    }
}
